src/Generators/Microsoft.Gen.Logging/Model/LoggingMethod.cs

```diff
@@ -28,8 +29,44 @@ internal sealed class LoggingMethod
     public bool LoggerMemberNullable;
     public bool HasXmlDocumentation;
 
-    public string GetParameterNameInTemplate(LoggingMethodParameter parameter)
-        => TemplateToParameterName.TryGetValue(parameter.Name, out var value)
-            ? value
-            : parameter.Name;
+    public LoggingMethodParameter? GetParameterForTemplate(string templateName)
+    {
+        foreach (var p in Parameters)
+        {
+            if (templateName.Equals(p.ParameterName, StringComparison.OrdinalIgnoreCase))
+            {
+                return p;
+            }
+        }
+
+        return null;
+    }
+
+    public List<string> GetTemplatesForParameter(LoggingMethodParameter lp)
```

Comment:
This overload can use the overload defined below:
```cs
public List<string> GetTemplatesForParameter(LoggingMethodParameter lp)
    => GetTemplatesForParameter(lp.ParameterName);
```

Reply:
Done.